scenarios/remote_call_rate/src/main.rs

```diff
@@ -0,0 +1,142 @@
+use anyhow::Context;
+use holochain_types::prelude::{AgentPubKey, ExternIO};
+use remote_call_integrity::TimedResponse;
+use std::time::{Duration, Instant};
+use trycp_wind_tunnel_runner::prelude::*;
+
+const CONDUCTOR_CONFIG: &str = include_str!("../../../conductor-config.yaml");
+
+#[derive(Debug, Default)]
+pub struct ScenarioValues {
+    remote_call_peers: Vec<AgentPubKey>,
+}
+
+impl UserValuesConstraint for ScenarioValues {}
+
+fn agent_setup(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<ScenarioValues>>,
+) -> HookResult {
+    connect_trycp_client(ctx)?;
+    reset_trycp_remote(ctx)?;
+
+    let client = ctx.get().trycp_client();
+    let agent_id = ctx.agent_id().to_string();
+
+    ctx.runner_context()
+        .executor()
+        .execute_in_place(async move {
+            client
+                .configure_player(agent_id.clone(), CONDUCTOR_CONFIG.to_string(), None)
+                .await?;
+
+            client
+                .startup(agent_id.clone(), Some("warn".to_string()), None)
+                .await?;
+
+            Ok(())
+        })?;
+
+    install_app(
+        ctx,
+        scenario_happ_path!("remote_call"),
+        &"remote_call".to_string(),
+    )?;
+    try_wait_for_min_peers(ctx, Duration::from_secs(120))?;
+
+    Ok(())
+}
+
+fn agent_behaviour(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<ScenarioValues>>,
+) -> HookResult {
+    let client = ctx.get().trycp_client();
+
+    let agent_id = ctx.agent_id().to_string();
+    let app_port = ctx.get().app_port();
+    let cell_id = ctx.get().cell_id();
+    let next_remote_call_peer = ctx.get_mut().scenario_values.remote_call_peers.pop();
+    let reporter = ctx.runner_context().reporter();
+
+    let new_peers = ctx
+        .runner_context()
+        .executor()
+        .execute_in_place(async move {
+            match next_remote_call_peer {
+                None => {
+                    // No more agents available to call, get a new list.
+                    // This is also the initial condition.
+                    Ok(client
+                        .agent_info(agent_id, None, None)
+                        .await?
+                        .into_iter()
+                        .map(|info| AgentPubKey::from_raw_36(info.agent.0.clone()))
+                        .filter(|k| k != cell_id.agent_pubkey()) // Don't call ourselves!
+                        .collect::<Vec<_>>())
```

Comment:
`.shuffle()`?

Reply:
Yes, that makes sense. It's possible that because we start peers with some ordering, that they discover each other with some ordering and end up with several making calls to each other at the same time - evening that out is a good idea